awx/main/scheduler/task_manager_models.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2022 Ansible by Red Hat
+# All Rights Reserved.
+import logging
+
+from django.conf import settings
+
+from awx.main.models import (
+    Instance,
+    InstanceGroup,
+)
+
+logger = logging.getLogger('awx.main.scheduler')
+
+
+class TaskManagerInstance:
+    """A class representing minimal data the task manager needs to represent an Instance."""
+
+    def __init__(self, obj, node_type, capacity, hostname):
+        self.obj = obj
+        self.node_type = node_type
+        self.remaining_capacity = capacity
+        self.capacity = capacity
+        self.hostname = hostname
+
+
+class TaskManagerInstances:
+    def __init__(self, active_tasks=[]):
+        instances = Instance.objects.filter(hostname__isnull=False, enabled=True).exclude(node_type='hop').only('node_type', 'capacity', 'hostname', 'enabled')
+        self._remaining_capacity_updated = False
+        instance_list = [
+            TaskManagerInstance(
+                obj=instance,
+                node_type=instance.node_type,
+                capacity=instance.capacity,
+                hostname=instance.hostname,
+            )
+            for instance in instances
+        ]
+        self._update_remaining_capacity(active_tasks)
+        self.instances_by_hostname = {i.hostname: i for i in instance_list}
+
+    def _update_remaining_capacity(self, tasks):
+        """Takes a list of tasks and updates the remaining capacity on our TaskManagerInstance list.
+
+        Computes remaining capacity for all the instances based on currently running and waiting tasks.
+        """
+        if self._remaining_capacity_updated:
+            # Only want to do this once
```

Comment:
Yeah, I just about add a comment in your last PR criticizing this, but you're wrapping that loose end up here.